tests/common/test_path.py

```diff
@@ -193,3 +210,193 @@ def test_path_identity(tmp_path: Path) -> None:
+        pytest.param("////", "{LIB}\\", None, id="root"),
+        pytest.param(None, "{ALT}\\", "/", id="root"),
+        pytest.param("/root", "{LIB}\\root", True, id="root"),
+        pytest.param("///root", "{LIB}\\root", None, id="root"),
+        pytest.param("////root", "{LIB}\\root", None, id="root"),
+        pytest.param(None, "{ALT}\\root", "/root", id="root"),
+        pytest.param("/root/", "{LIB}\\root\\", True, id="root"),
+        pytest.param("///root/", "{LIB}\\root\\", None, id="root"),
+        pytest.param("////root/", "{LIB}\\root\\", None, id="root"),
+        pytest.param(None, "{ALT}\\root\\", "/root/", id="root"),
+        pytest.param("/root/CaSe", "{LIB}\\root\\CaSe", True, id="root"),
+        pytest.param("///root/CaSe", "{LIB}\\root\\CaSe", None, id="root"),
+        pytest.param(None, "{ALT}\\root\\CaSe", "/root/CaSe", id="root"),
+        # UNC mount (2 leading slashes)
+        pytest.param("//", "\\\\", True, id="UNC"),
+        pytest.param("//mount", "\\\\mount", True, id="UNC"),
+        pytest.param("//mount/", "\\\\mount\\", True, id="UNC"),
+        pytest.param("//mount//", "\\\\mount\\", None, id="UNC"),
+        pytest.param(None, "\\\\mount\\", "//mount/", id="UNC"),
+        pytest.param("//mount/CaSe", "\\\\mount\\CaSe", True, id="UNC"),
+        pytest.param("//mount//CaSe", "\\\\mount\\CaSe", None, id="UNC"),
+        pytest.param(None, "\\\\mount\\CaSe", "//mount/CaSe", id="UNC"),
+        # drive (1 leading slash + 1 letter)
+        # /c & /C doesn't roundtrip
+        pytest.param("/c", "C:\\", None, id="drive"),
+        pytest.param("/C", "C:\\", None, id="drive"),
+        # c: & C: doesn't roundtrip
+        pytest.param(None, "c:", "/c", id="drive"),
+        pytest.param(None, "C:", "/c", id="drive"),
+        pytest.param("/c/", "C:\\", True, id="drive"),
+        pytest.param("/C/", "C:\\", None, id="drive"),
+        pytest.param("/c//", "C:\\", None, id="drive"),
+        pytest.param("/C//", "C:\\", None, id="drive"),
+        pytest.param(None, "c:\\", "/c/", id="drive"),
+        pytest.param(None, "C:\\", "/c/", id="drive"),
+        pytest.param("/c/drive", "C:\\drive", True, id="drive"),
+        pytest.param(None, "c:\\drive", "/c/drive", id="drive"),
+        pytest.param(None, "C:\\drive", "/c/drive", id="drive"),
+        pytest.param(None, "c:\\drive", "/c/drive", id="drive"),
+        pytest.param("/c/drive/CaSe", "C:\\drive\\CaSe", True, id="drive"),
+        pytest.param(None, "c:\\drive\\CaSe", "/c/drive/CaSe", id="drive"),
+        # relative path
+        pytest.param("relative", "relative", True, id="relative"),
+        pytest.param("relative/", "relative\\", True, id="relative"),
+        pytest.param("relative/CaSe", "relative\\CaSe", True, id="relative"),
+        # odd cases
+        pytest.param("path:", "path;.", None, id="colon"),
+        pytest.param(None, "path;.", "path:.", id="colon"),
+        pytest.param(None, "path;", "path", id="colon"),
```

Comment:
I would add some variants where the "blank" is at the start, `":path"` etc.

Reply:
good idea, added in https://github.com/conda/conda/pull/14157/commits/f93c1babef0fa5ad83be4c4c23b7b6c4a5d445ca